### DISCUSSION TITLE
Form的数据在页面切换时不会彻底清空

### DISCUSSION THREAD

**Discussion by lppoi** [alibaba/formily#3957] [Q&A 2.x] [answered]
我的编辑和新增公用的一个页面组件，点进编辑页时通过form.setValues填充数据（内含不是表单项的数据，如id,create_time等字段），返回上一页后进新增页时表单数据还在，如何彻底清空或者卸载表单组件。
```js
useEffect(() => {
    return () => {
      formStep.setCurrent(0);
      form.reset()
    };
  }, []);
useEffect(() => {
    if (params.id) {
      setLoading(true);
      getSampleByIdAPI(params.id).then((res) => {
        form.setValues(res);
        setLoading(false);
      });
    }
  }, [params]);
```

**Comment by lppoi** [comment]
![image](https://github.com/alibaba/formily/assets/64509267/ddec3ceb-f9d6-44e8-98ee-c657ac92a725)
表单代码

**Comment by liudichen** [comment, marked as answer]
const form = useMemo(()=>createForm({initialValues}),[initialValues])，编辑或新增时重新生成form实例可能就可以了吧，不共用form实例

**Reply by lppoi** [reply]
是的 我第二天发现问题了，form实例写在了组件外层，所以导致了切换页面状态不清空
